app/views/repositories/_repository.html.erb

```diff
@@ -34,11 +34,11 @@
            data-toggle="dropdown"
            aria-haspopup="true"
            aria-expanded="true"
-           <%= "disabled='disabled'" unless can_manage_repository?(repository.team) %>>
+           <%= "disabled='disabled'" if !can_update_or_delete_repository?(repository) and !can_create_repositories?(repository.team) %>>
         <span class="glyphicon glyphicon-cog"></span>
         <span class="caret"></span>
       </div>
-      <% if can_manage_repository?(repository.team) %>
+      <% if can_update_or_delete_repository?(repository) or can_create_repositories?(repository.team) %>
```

Comment:
Same here, use `||` instead of `or`. `and` and `or` should be used for control flow, not boolean operations [[see this]](http://www.virtuouscode.com/2010/08/02/using-and-and-or-in-ruby/). Of course, we previously also used them wrongly, especially in scope of permissions.